src/Simplify.elm

```diff
@@ -4152,12 +4154,17 @@ listFoldAnyDirectionChecks foldOperationName checkInfo =
 
                     else if isBinaryOperation "*" checkInfo checkInfo.firstArg then
                         if initialNumber == Just 0 then
-                            [ Rule.errorWithFix
-                                { message = "The call to List." ++ foldOperationName ++ " (*) 0 will result in 0."
-                                , details = [ "You can replace this call by 0." ]
+                            [ Rule.error
```

Comment:
I was thinking that the error message here didn't fit as nicely as in the other places.

But I'm thinking we maybe should not check for `initialNumber == Just 0` anymore in this situation. That way, we'll end up with `initialNumber * List.product list`. If `initialNumber` happens to be 0, then we'll end up with `0 * List.product list`, which will trigger the error in other places.
What do you think?